coilsnake/modules/eb/StaffModule.py

```diff
@@ -0,0 +1,268 @@
+
+                if mark == '>':
+                    self.read_space_from_project(int(text))
+                elif mark not in ['#', '-']:
+                    self.read_keyword_from_project(mark + text)
+                elif len(text) > SCREEN_WIDTH_IN_TILES:
+                    raise InvalidUserDataError(
+                        'Text \'{}\' is too long - must have at most {} characters'.format(
+                            text, SCREEN_WIDTH_IN_TILES
+                        )
+                    )
+                elif mark == '#':
+                    self.read_small_line_from_project(text)
+                else:
+                    self.read_big_line_from_project(text)
+
+        self.data.append(CONTROL_END_OF_STAFF)
+
+    def print_keyword(self, f, byte):
+        for item in KEYWORD_BYTE_HEIGHT.items():
+            keyword, v = item
+            b, h = v
+
+            if byte == b:
+                f.write('{}\n'.format(keyword))
+                return MODE_CONTROL
+        else:
+            raise CoilSnakeError('Unknown control byte 0x{:X}'.format(byte))
+
+    def print_control(self, f, byte):
+        if byte == CONTROL_SMALL:
+            f.write('# ')
+            return MODE_SMALL
+        elif byte == CONTROL_BIG:
+            f.write('- ')
+            return MODE_BIG
+        elif byte == CONTROL_SPACE:
+            f.write('> ')
+            return MODE_SPACE
+        elif byte == CONTROL_END_OF_STAFF:
+            return MODE_CONTROL
+        else:
+            return self.print_keyword(f, byte)
+
+    def print_space(self, f, byte):
+        f.write('{}\n\n'.format(byte))
+        return MODE_CONTROL
+
+    def print_char(self, f, byte, mode, mapping_big, mapping_small):
+        if byte == CONTROL_END_OF_LINE:
+            f.write('\n')
+            return MODE_CONTROL
+
+        if mode == MODE_BIG:
+            f.write(mapping_big[byte])
+        else:
+            f.write(mapping_small[byte])
+
+        return mode
+
+    def read_pointer_from_rom(self, rom):
+        return (
+            (rom[STAFF_TEXT_POINTER_OFFSET + 6] << 16) |
+            rom.read_multi(STAFF_TEXT_POINTER_OFFSET, 2)
+        )
+
+    def write_pointer_to_rom(self, rom, pointer):
+        rom[STAFF_TEXT_POINTER_OFFSET + 6] = pointer >> 16
+        rom.write_multi(STAFF_TEXT_POINTER_OFFSET, pointer & 0xFFFF, 2)
+
+    def read_from_rom(self, rom):
+        staff_text_offset = from_snes_address(self.read_pointer_from_rom(rom))
+        self.read_staff_chars_from_assets()
+        log.debug('Reading staff text')
+        i    = 0
+        byte = 0x00
+
+        while byte != CONTROL_END_OF_STAFF:
+            byte = rom[staff_text_offset + i]
+            self.data.append(byte)
+            i += 1
+
+    def write_to_rom(self, rom):
+        offset = rom.allocate(size=len(self.data))
+        self.write_pointer_to_rom(rom, to_snes_address(offset))
+        log.debug('Writing staff text')
+
+        for byte in self.data:
```

Comment:
IIRC rom (and all blocks in general) supports some kind of syntax like `rom[offset:offset+len(data)] = self.data`

Reply:
Yep. Done.